test/e2e_env/universal/timeout/envoyconfig.go

```diff
@@ -0,0 +1,101 @@
+package timeout
+
+import (
+	"encoding/json"
+	"os"
+	"regexp"
+	"strings"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	mesh_proto "github.com/kumahq/kuma/api/mesh/v1alpha1"
+	"github.com/kumahq/kuma/pkg/test"
+	"github.com/kumahq/kuma/pkg/test/matchers"
+	"github.com/kumahq/kuma/pkg/test/resources/builders"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/framework/client"
+	"github.com/kumahq/kuma/test/framework/envs/universal"
+)
+
+func EnvoyConfigTest() {
+	meshName := "meshtimeout-envoyconfig"
+
+	BeforeAll(func() {
+		err := NewClusterSetup().
+			Install(
+				Yaml(
+					builders.Mesh().
+						WithName(meshName).
+						WithoutInitialPolicies().
+						WithMeshServicesEnabled(mesh_proto.Mesh_MeshServices_Exclusive),
+				),
+			).
+			Install(DemoClientUniversal("demo-client", meshName,
+				WithTransparentProxy(true)),
+			).
+			Install(TestServerUniversal("test-server", meshName,
+				WithArgs([]string{"echo", "--instance", "universal-1"})),
+			).
+			Setup(universal.Cluster)
+		Expect(err).ToNot(HaveOccurred())
+
+		Eventually(func(g Gomega) {
+			_, err := client.CollectEchoResponse(universal.Cluster, "demo-client", "test-server.svc.mesh.local")
+			g.Expect(err).ToNot(HaveOccurred())
+		}).Should(Succeed())
+	})
+
+	AfterEachFailure(func() {
+		DebugUniversal(universal.Cluster, meshName)
+	})
+
+	E2EAfterAll(func() {
+		Expect(universal.Cluster.DeleteMeshApps(meshName)).To(Succeed())
+		Expect(universal.Cluster.DeleteMesh(meshName)).To(Succeed())
+	})
+
+	E2EAfterEach(func() {
+		// delete all meshtimeout policies
+		out, err := universal.Cluster.GetKumactlOptions().RunKumactlAndGetOutput("get", "meshtimeouts", "--mesh", meshName, "-o", "json")
+		Expect(err).ToNot(HaveOccurred())
+		var output struct {
+			Items []struct {
+				Name string `json:"name"`
+			} `json:"items"`
+		}
+		Expect(json.Unmarshal([]byte(out), &output)).To(Succeed())
+		for _, item := range output.Items {
+			Expect(universal.Cluster.GetKumactlOptions().RunKumactl("delete", "meshtimeout", item.Name, "--mesh", meshName)).To(Succeed())
+		}
+	})
+
+	redactIPs := func(jsonStr string) string {
```

Comment:
can we extract this to some common fuctions, as we would like to use this in other similar tests for policies?

Reply:
If we make this test more general, then probably this comment does not make sense